superset/utils/log.py

```diff
@@ -83,6 +86,43 @@ def stats_logger(self):
         return current_app.config.get("STATS_LOGGER")
 
 
+def get_event_logger_from_cfg_value(cfg_value: object) -> AbstractEventLogger:
+    """
+    This function implements the deprecation of assignment of class objects to EVENT_LOGGER
+    configuration, and validates type of configured loggers.
+
+    The motivation for this method is to gracefully deprecate the ability to configure
+    EVENT_LOGGER with a class type, in favor of preconfigured instances which may have
+    required construction-time injection of proprietary or locally-defined dependencies.
+
+    :param cfg_value: The configured EVENT_LOGGER value to be validated
+    :return: if cfg_value is a class type, will return a new instance created using a
+    default con
+    """
+    result: Any = cfg_value
+    if inspect.isclass(cfg_value):
+        logging.getLogger().warning(
+            """
+            In superset private config, EVENT_LOGGER has been assigned a class object. In order to
+            accomodate pre-configured instances without a default constructor, assignment of a class
+            is deprecated and may no longer work at some point in the future. Please assign an object
+            instance of a type that implements superset.utils.log.AbstractEventLogger.
+            """
```

Comment:
This should be wrapped in `textwrap.dedent` (see an example here: https://github.com/apache/incubator-superset/blob/d65b039219c9825ad50ec03ad73a1638710c73c9/superset/examples/css_templates.py#L30).